### PR TITLE
fix: parse all colors to rgba objects instead of numbers

### PR DESCRIPTION
**Pull request by KaelWD**
fixes #16045

### CODE REVIEW

File: packages/vuetify/src/util/colorUtils.ts [hunk 1]
```diff
@@ -75,24 +76,8 @@ export function classToHex (
...
-}
-
-/**
- * Converts HSVA to RGBA. Based on formula from https://en.wikipedia.org/wiki/HSL_and_HSV
- *
- * @param color HSVA color as an array [0-360, 0-1, 0-1, 0-1]
- */
-export function HSVAtoRGBA (hsva: HSVA): RGBA {
+/** Converts HSVA to RGBA. Based on formula from https://en.wikipedia.org/wiki/HSL_and_HSV */
+export function HSVAtoRGBA (hsva: HSV): RGB {
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/util/colorUtils.ts:80]: `HSVtoRGB`?

File: packages/vuetify/src/util/colorUtils.ts [hunk 2]
```diff
@@ -158,49 +139,41 @@ export function HSLAtoHSVA (hsl: HSLA): HSVA {
   return { h, s: sprime, v, a }
 }
 
-export function RGBAtoCSS (rgba: RGBA): string {
+export function RGBAtoCSS (rgba: RGB): string {
   return `rgba(${rgba.r}, ${rgba.g}, ${rgba.b}, ${rgba.a})`
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/util/colorUtils.ts:143]: won't it end up with `rgba(N, N, N, )` if `.a` is undefined?

File: packages/vuetify/src/components/VColorPicker/VColorPicker.tsx [hunk 3]
```diff
@@ -101,7 +101,7 @@ export const VColorPicker = defineComponent({
       }
     )
 
-    const updateColor = (hsva: HSVA) => {
+    const updateColor = (hsva: HSV) => {
```
Comment (johnleider) [inline packages/vuetify/src/components/VColorPicker/VColorPicker.tsx:104]: should the variable name be hsv?

File: packages/vuetify/src/components/VColorPicker/util/index.ts [hunk 4]
```diff
@@ -1,41 +1,37 @@
...
 function has (obj: object, key: string[]) {
   return key.every(k => obj.hasOwnProperty(k))
 }
 
-export function parseColor (color: any): HSVA | null {
+export function parseColor (color: any): HSV | null {
   if (!color) return null
 
-  let hsva: HSVA | null = null
+  let hsva: HSV | null = null
```
Comment (johnleider) [inline packages/vuetify/src/components/VColorPicker/util/index.ts:22]: should variable name be hsv?

File: packages/vuetify/src/components/VColorPicker/util/index.ts [hunk 5]
```diff
@@ -54,9 +50,9 @@ function stripAlpha (color: any, stripAlpha: boolean) {
   return color
 }
 
-export function extractColor (color: HSVA, input: any) {
+export function extractColor (color: HSV, input: any) {
```
Comment (johnleider) [inline packages/vuetify/src/components/VColorPicker/util/index.ts:53]: small nitpick, should the color variable be hsv to match other naming conventions?

File: packages/vuetify/src/util/colorUtils.ts [hunk 6]
```diff
@@ -7,50 +7,51 @@ import * as CIELAB from '@/util/color/transformCIELAB'
...
-export type HSVA = HSV & { a: number }
-export type RGB = { r: number, g: number, b: number }
-export type RGBA = RGB & { a: number }
-export type HSL = { h: number, s: number, l: number }
-export type HSLA = HSL & { a: number }
-export type Hex = string
-export type Hexa = string
+export type HSV = { h: number, s: number, v: number, a?: number }
+export type RGB = { r: number, g: number, b: number, a?: number }
+export type HSL = { h: number, s: number, l: number, a?: number }
```
Comment (johnleider) [inline packages/vuetify/src/util/colorUtils.ts:14]: should a?: number be here?

**Review by johnleider — CHANGES_REQUESTED** [review]
I didn't mark all of the instances of **a** existing but you get the point.

**Review by johnleider — APPROVED** [review]

**Pull request merged**